### DISCUSSION TITLE
Is there documentation to run FluxCD behind a firewall?

### DISCUSSION THREAD

**Discussion by GCHQ77703** [fluxcd/flux2#913] [Q&A]
We run all our applications behind a firewall, meaning that the installation process of FluxCD is a little tricky.  It expects there to be the internet in several cases.  Is there any documentation for installing on a system without direct access to the internet (but where binaries / code can be imported manually using e.g. a USB stick)?

**Comment by stefanprodan** [comment]
After Go 1.6 gets released, we will use the new [embed package](https://github.com/golang/go/issues/41191) to bundle the components manifests inside `flux` CLI binary. Then you'll be able to copy the binary and it will no longer try to connect to GitHub. As for the container images, they are hosted on GitHub Container Registry, you'll need to pull them from there and push them to your private registry, then you can [supply your registry](https://toolkit.fluxcd.io/guides/installation/#generic-git-server) to flux install.

**Reply by stefanprodan** [reply]
PR here: https://github.com/fluxcd/flux2/pull/987
